pkg/image/registryclient/client.go

```diff
@@ -223,7 +224,12 @@ func (c *Context) Repository(ctx context.Context, registry *url.URL, repoName st
 	if limiter == nil {
 		limiter = rate.NewLimiter(rate.Limit(5), 5)
 	}
-	return NewLimitedRetryRepository(repo, c.Retries, limiter), nil
+	repo = NewLimitedRetryRepository(repo, c.Retries, limiter)
+	manifests, err := repo.Manifests(ctx)
```

Comment:
I'm confused why this was added?  Manifests isn't required to be returned up front, it is a lazy initialization.

Reply:
right, I can remove it, i left it because it saves a lot of code elsewhere

Reply:
updated